apps/settings/js/datasync/history_adapter.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+/* globals IAC */
+/* exported SynctoHistoryAdapter */
+
+var HistoryAPI = {
+  addPlaces: function(places) {
+    return IAC.request('sync-history', {
+      method: 'addPlaces',
+      args: [places]
+    });
+  },
+
+  addPlace: function(place) {
+    return IAC.request('sync-history', {
+      method: 'addPlace',
+      args: [place]
+    });
+  }
+};
+
+var SynctoHistoryAdapter = {
+  update: function(kintoCollection) {
+    function updateHistoryCollection(list) {
+      var historyRecords = list.data;
+      var places = [];
+      historyRecords.forEach(function (decryptedRecord){
+        var record = decryptedRecord.payload;
+        if(!record.histUri || !record.visits || !record.visits[0]){
+          return ;
+        }
+
+        var visits = [];
+        record.visits.forEach(function (elem){
+          visits.push(elem.date);
+        });
+
+        var place = {
+          url: record.histUri,
+          title: record.title,
+          visits: visits,
+          last_modified: decryptedRecord.last_modified
+        };
+        places.push(place);
+      });
+
+      return HistoryAPI.addPlaces(places);
+    }
+
+    return kintoCollection.list().then(list => {
```

Comment:
for kintoCollection.list() part, I would like to use a function to wrapper updateHistoryCollection again. That's more convenient for using debugger.

for function or =>, I would like to use arrow function as well, so all anonymous functions have been modified to arrow functions.